proposed/2021/VisualStudioPackageInstallDeepLink.md

```diff
@@ -0,0 +1,163 @@
+# Visual Studio Package Install Deep Link
+
+- Author Name: Chris Gill ([chgill-MSFT](https://github.com/chgill-MSFT))
+- Start Date: 07/19/2021
+- GitHub Issue: https://github.com/NuGet/Home/issues/11032
+- GitHub PR: N/A
+
+## Summary
+
+Support deep links for NuGet.org packages that, when clicked, open the package in the Visual Studio NuGet Package Manager UI (PMUI) for a 1 click install experience.
+
+Applications:
+
+* Add “Install with Visual Studio” link to package README, documentation, or tutorials
+* Deep link suggested alternative package for deprecated packages in Visual Studio for an easier transition experience
+* Share install link with coworkers
+* Include “Install with Visual Studio” option on NuGet.org
+
+## Motivation 
+
+Today, customers must copy a CLI command to install a NuGet package to their project from NuGet.org, or search for the package in Visual Studio. We should enable customers to directly install the package from the Visual Studio package manager UI so they can continue to use their preferred package management tool.
+
+From our NuGet.org survey, 70% of our respondents reported using the Visual Studio NuGet UI to install packages. Additionally, 87% of our respondents use Visual Studio for developing .NET apps. 
+
+Since users prefer installing packages via the VS package manager UI by a large margin relative to CLI methods, we should support that as a first-class experience.
+
+## Success metric
+
+We will use telemetry the measure % of VS package installs that come from a deep link and surveys to understand satisfaction with available installation options for NuGet packages.
+
+## Scope
+
+* Generate a deep link URL that opens the NuGet package in the Visual Studio PMUI
+* Handler is just for Visual Studio
+* Only NuGet.org packages are supported
+
+## Scenarios
+
+1.	Visual Studio isn’t open
+2.	Visual Studio is open without a solution selected
+3.	A solution is open, but the PMUI isn’t
+4.	Project level PMUI is open
+5.	Solution level PMUI is open
+
+## Design
+
+### Deep link design
+
+* Must specify package ID
+* Can specify version
+* No version specified defaults to latest version available
+* Structure: TBD
+
+* Example from VS Code extension deep link: [vscode:extension/dracula-theme.theme-dracula](vscode:extension/dracula-theme.theme-dracula)
+
+### Scenario 1 - Visual Studio isn’t open
+
+1. Deep link URL is clicked.
+2. Open Visual Studio to solution selector window and wait for solution selection
```

Comment:
I wonder how frequent the solution selector window is used/known to most VS users...this *may* be a jarring experience.

Reply:
Don't customers see it every time they launch VS to get to their solution? At least I don't typically click on the .sln file directly since it's dug deep in some folder somewhere. I usually launch VS from my taskbar and choose the solution from the selection menu.

Reply:
Ahhhh. I thought you were referring to an older dialog.  If you are referring to the dialog that is this:
![image](https://user-images.githubusercontent.com/4821/126552747-bdad6836-5f84-47d2-8e61-3bd1fa91b6cd.png)
That makes sense. I've never heard anyone call this that but makes more sense now :-)